coredns.yaml

```diff
@@ -0,0 +1,42 @@
+package:
+  name: coredns
+  version: 1.10.1
+  epoch: 0
+  description: CoreDNS is a DNS server that chains plugins
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - ca-certificates-bundle
+
+environment:
+  contents:
+    packages:
+      - ca-certificates-bundle
+      - busybox
+      - bash
+      - make
+      - go
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/coredns/coredns
+      tag: v${{package.version}}
+      expected-commit: 055b2c31a9cf28321734e5f71613ea080d216cd3
+
+  - runs: |
+      # Ensures plugins get included
+      make check
+
+  - uses: go/build
+    with:
+      packages: .
+      output: coredns
+      ldflags: -s -w -X github.com/coredns/coredns/coremain.GitCommit=v${{package.version}}
```

Comment:
Do we need the setcap bit, or does that come later in apko? https://github.com/coredns/coredns/blob/master/Dockerfile#L13

Reply:
that's a great question, I don't know... I don't even know if melange's solver will track/keep that? I also don't know if apko supports `setcap`ing.

do we have precedence for this anywhere?

Reply:
No, I think this package was blocked on some recent work to make setcap work somehow so this will likely be the first to use it! cc @kaniini @deitch

Reply:
I think we need to update melange to use a newer version of the go-apk component first.